gpflow/quadrature/gauss_hermite.py

```diff
@@ -0,0 +1,110 @@
+from typing import List
+
+import numpy as np
+import tensorflow as tf
+
+from .base import GaussianQuadrature
+from ..config import default_float
+
+from ..base import TensorType
+
+
+def gh_points_and_weights(n_gh: int):
+    r"""
+    Given the number of Gauss-Hermite points n_gh,
+    returns the points z and the weights dz to perform the following
+    uni-dimensional gaussian quadrature:
+
+    X ~ N(mean, stddev²)
+    E[f(X)] = ∫f(x)p(x)dx = sum_{i=1}^{n_gh} f(mean + stddev*z_i)*dz_i
+
+    :param n_gh: Number of Gauss-Hermite points, integer
+    :returns: Points z and weights dz, both tensors with shape [n_gh],
+        to compute uni-dimensional gaussian expectation
+    """
+    z, dz = np.polynomial.hermite.hermgauss(n_gh)
+    z = z * np.sqrt(2)
+    dz = dz / np.sqrt(np.pi)
+    z, dz = z.astype(default_float()), dz.astype(default_float())
+    return tf.convert_to_tensor(z), tf.convert_to_tensor(dz)
+
+
+def list_to_flat_grid(xs: List[TensorType]):
+    """
+    :param xs: List with d rank-1 Tensors, with shapes N1, N2, ..., Nd
+    :return: Tensor with shape [N1*N2*...*Nd, dim] representing the flattened
+        D-dimensional grid built from the input tensors xs
+    """
+    return tf.reshape(tf.stack(tf.meshgrid(*xs), axis=-1), (-1, len(xs)))
+
+
+def reshape_Z_dZ(zs: List[TensorType], dzs: List[TensorType]):
+    """
+    :param zs: List with d rank-1 Tensors, with shapes N1, N2, ..., Nd
```

Comment:
are N1, ..., Nd all equal to n_gh here? I would assume so given the shape of the return value?

Reply:
In the current implementation, yes, they are equal.
However, `reshape_Z_dZ` can also covers grids with different resolution for each latent function, hence why specifying the shapes that way.
In fact, I will fix the docsting for the return type. Instead of:
```
    :returns: points Z, Tensor with shape [n_gh**dim, dim],
        and weights dZ, Tensor with shape [n_gh**dim, 1]
```
It should be
```
    :returns: points Z, Tensor with shape [N1*N2*...*Nd, d],
        and weights dZ, Tensor with shape [N1*N2*...*Nd, 1]
```

Reply:
yes - perfect.

Reply:
fixed in a69acda